src/context/auth.tsx

```diff
@@ -0,0 +1,142 @@
+import React, {
```

Comment:
This is the majority of the change. This context module provides access to the `AuthProvider` and the supporting custom hooks.